File: 3169-count-days-without-meetings/3169-count-days-without-meetings.py
class Solution:
    def merge(self, A):
        A.sort()
        res = [A[0]]
        for i in range(1, len(A)):
            if res[-1][1] >= A[i][0]:
                res[-1][1] = max(res[-1][1], A[i][1])
            else:
                res.append(A[i])
        return res

    def countDays(self, days, A):
        newA = self.merge(A)
        for interval in newA:
            days -= (interval[1] - interval[0] + 1)
        return days

File: 3169-count-days-without-meetings/3169-count-days-without-meetings.java
class Solution {
    TreeMap<Integer, Integer> intervalTree = new TreeMap<>();
    int totalCoveredLength = 0;

    public int countDays(int days, int[][] meetings) {
        for (int i = 0; i < meetings.length; i++) {
            var from = meetings[i][0];
            var to = meetings[i][1];
            Map.Entry<Integer, Integer> entry = intervalTree.floorEntry(to);
            while (entry != null && entry.getValue() >= from) {
                int existingLeft = entry.getKey();
                int existingRight = entry.getValue();
                totalCoveredLength -= existingRight - existingLeft + 1;
                from = Math.min(existingLeft, from);
                to = Math.max(existingRight, to);
                // Remove existing one
                intervalTree.remove(existingLeft);
                // Initialize entry to next one lesser than newly created entry
                entry = intervalTree.floorEntry(to);
            }
            // At the end insert merged interval and update total count
            intervalTree.put(from, to);
            totalCoveredLength += to - from + 1;
        }

        return days - totalCoveredLength;
    }

}